arvo/reference/cryptography.md

```diff
@@ -0,0 +1,222 @@
++++
+title = "Cryptography"
+weight = 10
+template = "doc.html"
++++
+
+This is a summary of the cryptography functions found in `sys/zuse.hoon`, Arvo's
+standard library. This page currently only documents cryptographic functions
+directly utilized by [Ames](@/docs/arvo/ames/ames.md). `zuse` also contains
+cryptographic functions relevant to Ethereum such as the `+keccak:crypto` core,
+but they are currently undocumented.
+
+Documentation for [Insecure Hashing](@/docs/hoon/reference/stdlib/2e.md) and the
+[SHA Hash Family](@/docs/reference/stdlib/3d.md) is found in the Hoon standard
+library reference.
+
+## Summary
+
+`zuse` contains several cryptosuites. The ones utilized by Ames are
+[`+ed:crypto`](#ed), [`+aes:crypto`](#aes), and [`+crub:crypto`](#crub), with
+the latter being the only one which is implementable as an [`+acru:ames`
+core](@/docs/arvo/ames/cryptography/#crypto-core).
+
+## `+crub:crypto` {#crub}
+
+`+crub:crypto` implements an
+[`+acru:ames`](@/docs/arvo/ames/cryptography/#crypto-core) core that implements
+[Suite B Cryptography](https://en.wikipedia.org/wiki/NSA_Suite_B_Cryptography).
+
+It utilizes AES symmetric key encryption and decryption from [`+aes:crypto`](#aes)
+implemented using the Diffie-Hellman key exchange protocol, elliptic curve
+digital signature algorithm (ECDSA) signing and verification with [`+ed:crypto`](#ed),
+and generates public/private key pairs using elliptic curve cryptography with
+`+ed:crypto`.
+
+A `+crub:crypto` core's payload contains public encryption and authentication
+keys and optional secret encryption and authentication keys.
+```hoon
+  ++  crub  !:
+    ^-  acru
+    =|  [pub=[cry=@ sgn=@] sek=(unit [cry=@ sgn=@])]
+    |%
+    ...
+```
+`+crub` cores are typically created using one of the constructors in [`+nu:crub`](#nu).
```

Comment:
```suggestion
`+crub` cores (because they follow the `+acru` interface) are typically created using one of the constructors in [`+nu:crub`](#nu).
```

Reply:
You could consider scrapping the explanations here entirely in favor of referring to the `+acru` docs. (And/or moving the explanations from here, over there.)

Reply:
Since `+crub` lives in `zuse.hoon` I don't want to move it to the Ames section. I generally think it is good to avoid duplicating information in reference sections, but I'm not sure if it should "count" if one section is describing an interface and the other is describing an implementation of that interface. Though it becomes even more of an edge case when there is only one implementation of an interface that is ever used, like it is here. I'm inclined to leave it as-is, but I'm going to check out how other projects split up documentation for interfaces/implementations and see if its done similarly or otherwise.